### PR TITLE
Mini_Message_Board

### PR DESCRIPTION
**Pull request by aniketrathi**
## Description

GET/
### Description
Show all messages

Response:
200

```
index.ejs from view

```

GET/POST /new
### Description:
GET- Form for adding messages
Response:
200

```
new.ejs from view

```
POST- add messages

Request Body:
```
{
  user: ' user name',
  text: 'message',
  added: 'current date'
}

```
Response:
200

```
index.ejs from view

```

## Database schema changes
- None

### Automated test cases added
- None

### Manual test cases run
- None

### CODE REVIEW

File: messages.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+      added: new Date(),
+    },
+    {
+      text: "Hello World!",
+      user: "Charles",
+      added: new Date(),
+    },
+  ];
+  module.exports = messages;
+
```
Comment (GaurisankarJ) [inline messages.js]: Remove the extra new line.

File: routes/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const express = require("express");
+const messages = require("../messages");
+const router = express.Router();
```
Comment (GaurisankarJ) [inline routes/index.js:4]: It's good to use more indentation, use blank spaces to separate different chunks of code.

File: routes/index.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    ...req.body,
+    added: new Date(),
+  };
+
+  if (!newmsg.user || !newmsg.text) {
+    return res.status(400).json({ msg: "Please include a name and text" });
+  }
+
+  messages.push(newmsg);
+  //res.json(messages);
```
Comment (GaurisankarJ) [inline routes/index.js]: Why is this commented?

Please remove if not required.

File: bin/www [hunk 4]
```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env node
```
Comment (GaurisankarJ) [inline bin/www:1]: Is this file part of the generated boiler plate?

File: app.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+var cookieParser = require("cookie-parser");
```
Comment (GaurisankarJ) [inline app.js:1]: Is this file part of the generated boiler plate?

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (GaurisankarJ) [inline .gitignore]: Add package-lock.json too.

**Comment by GaurisankarJ** [comment]
Please use below format for the routes in the description,
REQUEST METHOD REQUEST URL
DESCRIPTION
REQUEST BODY
RESPONSE
Eg,

POST /user
Description:
Create user.

Request Body:
```
{
  name: 'name',
  pass: 'pass,
}
```
Response:
200
```
{
 id: ...,
 name: 'name',
 pass: 'pass,
}
```

File: app.js (hunk 5 above)
Comment (aniketrathi) [reply app.js:1]: yes

File: bin/www (hunk 4 above)
Comment (aniketrathi) [reply bin/www:1]: yes

File: routes/index.js (hunk 3 above)
Comment (aniketrathi) [reply routes/index.js]: ok

**Review by GaurisankarJ — APPROVED** [review]

**Pull request merged**